### PR TITLE
docs: sequence diagram for scan machinery

### PR DESCRIPTION
**Pull request by erikgb**
This adds a sequence diagram showing the interactions and order of things between the user, relevant Kubernetes API and the operator controllers.

Preview:

image

### CODE REVIEW

File: docs/scan-sequence.puml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+participant "Scan Job controller" as jc
+entity      Job      as job
+
+user -> pod : create
+activate pod 
+
+wc <<- pod : watch
+activate wc
+wc -> cis : create
+deactivate wc
```
Comment (mikaelol) [inline docs/scan-sequence.puml:18]:
```suggestion
deactivate wc
activate cis
```

Example:

Comment (bendikp) [reply docs/scan-sequence.puml:18]: Nice, it shows the lifecycle of the CIS better 👍

Comment (erikgb) [reply docs/scan-sequence.puml:18]: Thanks @mikaelol!

File: docs/scan-sequence.puml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+activate job
+
+|||
+
+jc <<- job: watch
+deactivate job
+activate jc
+jc -> cis : update status
+deactivate jc
+
```
Comment (bendikp) [inline docs/scan-sequence.puml:34]: Maybe also add what happens when the pod is deleted?
```suggestion
user -> pod : delete
pod -->> cis : delete (GC)
deactivate pod 
deactivate cis
```
image

File: README.md [hunk 3]
```diff
@@ -228,7 +228,8 @@ Using a simple `Pod`, with a single container, as an example:
 3. When a scan `Job` is completed, read the scan result from pod log of the scan `Job`,
    and update the `ContainerImageScan` status.
```
Comment (bendikp) [inline README.md:230]:
```suggestion
4. When the `Pod` is deleted, the `ContainerImageScan` is garbage collected.
```

File: docs/scan-sequence.puml (hunk 2 above)
Comment (erikgb) [reply docs/scan-sequence.puml:38]: Good suggestion! Updated, PTAL!

**Review by mikaelol — APPROVED** [review]

**Review by bendikp — APPROVED** [review]

**Pull request merged**